app/express/interceptors/session/Session.js

```diff
@@ -0,0 +1,123 @@
+const {
+  Logger,
+  ConfigManager: { getConfig },
+} = require('@dojot/microservice-sdk');
+const session = require('express-session');
+const createError = require('http-errors');
+
+const SessionStore = require('./SessionStore')(session);
+
+const logger = new Logger('backstage:express/interceptors/session/Session');
+
+const { session: sessionConfig } = getConfig('BS');
+
+/**
+ * Renew the access token through the refresh token if the current access token is expired
+ *
+ * @param {object} req
+ * @param {an instance of ../keycloak} keycloak instance of the Keycloak class
+ */
+const renewAccessTokenIfNecessary = async (req, keycloak) => {
+  logger.debug('renewAccessTokenIfNecessary: ...');
+  // checks if the current token is expired
+  if ((Date.now() > new Date(req.session.accessTokenExpiresAt).getTime())) {
+    logger.debug('renewAccessTokenIfNecessary: Getting a new token...');
+    const {
+      accessToken,
+      refreshToken,
+      refreshExpiresAt,
+      accessTokenExpiresAt,
+      sessionState,
+    } = await keycloak.getRequestsInstance().getTokenByRefreshToken(
+      req.session.realm,
+      req.session.refreshToken,
+    );
+
+    req.session.accessToken = accessToken;
+    req.session.refreshToken = refreshToken;
+    req.session.refreshExpiresAt = refreshExpiresAt;
+    req.session.accessTokenExpiresAt = accessTokenExpiresAt;
+    req.session.sessionState = sessionState;
+
+    logger.debug('renewAccessTokenIfNecessary: ...got a new token.');
+  }
+};
+
+
+/**
+ * Middleware responsible for creating a session and managing it
+ *
+ * @param {String[]} exceptionRoutes  Array with Patchs  that does not need an active session
+ *                                    to be accessed
+ * @param {an instance of ../keycloak} object.keycloak instance of the Keycloak class
+ * @param {an instance of ../redis} object.redis instance of the Redis class
+ */
+module.exports = ({
+  exceptionRoutes,
+  keycloak,
+  redis,
+}) => ({
+  name: 'session-express-interceptor',
+  middleware:
+  [session({
+    secret: sessionConfig.secret,
```

Comment:
This secret is used to encrypt the session ID, right?
If a secret is not defined, it may be interesting to leave a random value instead of a (well-known) fixed value. 
If they forget to change the secret, it could be a security breach. 

This was done at kafkaWS:
https://github.com/dojot/dojot/blob/master/subscription-engine/kafka-ws/app/Config.js#L27

The problem is that if the backstage had to escalate, then it would have to define a secret anyway, even so I would opt for a random value and leave a very large note in the documentation stating to define a secret in case the service needs to be escalated.
I think that scaling the backstage service would not be that common....

Reply:
Great idea! :D
Thanks

Reply:
When configManager was added in kafka-ws, it was lost.
It may be necessary to at least put it in the readme which must be unique for each deployment, and also in docker-compose/ansible as environment variable this with a message saying that it must be unique for deployment.

Reply:
dojot/dojot#2092